mlflow/pyfunc/__init__.py

```diff
@@ -872,7 +872,7 @@ def _create_model_downloading_tmp_dir(should_use_nfs):
 _MLFLOW_SERVER_OUTPUT_TAIL_LINES_TO_KEEP = 200
 
 
-def _convert_model_output_spec_to_spark_type(spec):
+def _cast_spec_to_spark_type(spec):
```

Comment:
```suggestion
def _cast_output_spec_to_spark_type(spec):
```

For input spec, this function does not fit with it. (e.g. for 2D array input)